src/pacasam/extractors/bd_ortho_vintage.py

```diff
@@ -44,64 +47,69 @@ class BDOrthoVintageExtractor(Extractor):
-        with rasterio.open(rgb_file) as rgb, rasterio.open(irc_file) as irc:
+        with rasterio.open(rgb_file) as rgb_open, rasterio.open(irc_file) as irc_open:
             for patch_info in single_file_sampling.itertuples():
                 split = getattr(patch_info, SPLIT_COLNAME)
                 patch_id = getattr(patch_info, PATCH_ID_COLNAME)
                 tiff_patch_path: Path = self.make_new_patch_path(patch_id=patch_id, split=split)
                 if tiff_patch_path.exists():
                     continue
-
                 patch_geometry = getattr(patch_info, GEOMETRY_COLNAME)
-                bbox = patch_geometry.bounds
-                width = bbox[2] - bbox[0]
-                height = bbox[3] - bbox[1]
-                assert width == height  # squares only
-                width_pixels = int(self.pixel_per_meter * width)
-                rgb_arr = extract_patch_as_geotiffs(rgb, patch_geometry, width_pixels)
-                irc_arr = extract_patch_as_geotiffs(irc, patch_geometry, width_pixels)
-                image_resolution = 1 / self.pixel_per_meter
-                options = {
-                    "driver": "GTiff",
-                    "count": 4,
-                    "dtype": rgb_arr.dtype,
-                    "transform": Affine(image_resolution, 0, bbox[0], 0, -image_resolution, bbox[3]),
-                    "crs": rgb.crs,
-                    "width": width_pixels,
-                    "height": width_pixels,
-                    "compress": "DEFLATE",
-                    "tiled": False,
-                    "bigtiff": "IF_SAFER",
-                    "nodata": None,
-                }
-                tmp_patch: tempfile._TemporaryFileWrapper = tempfile.NamedTemporaryFile(suffix=self.patch_suffix, prefix="extracted_patch")
-                collate_rgbnir_and_save(options, rgb_arr, irc_arr, tmp_patch)
+                tmp_patch = extract_rgbnir_patch_as_tmp_file(rgb_open, irc_open, BDORTHO_PIXELS_PER_METER, patch_geometry)
                 tiff_patch_path.parent.mkdir(parents=True, exist_ok=True)
                 shutil.copy(tmp_patch.name, tiff_patch_path)
 
 
+def extract_rgbnir_patch_as_tmp_file(rgb_open, irc_open, pixel_per_meter, patch_geometry):
+    """Extract both rgb and irc patch images and collate them into a temporary file."""
+    bbox = patch_geometry.bounds
+    width = bbox[2] - bbox[0]
+    height = bbox[3] - bbox[1]
+    assert width == height  # squares only
+    width_pixels = int(pixel_per_meter * width)
+    rgb_arr = extract_patch_as_geotiffs(rgb_open, patch_geometry, width_pixels)
+    irc_arr = extract_patch_as_geotiffs(irc_open, patch_geometry, width_pixels)
+    image_resolution = 1 / pixel_per_meter
+    options = {
+        "driver": "GTiff",
+        "count": 4,
+        "dtype": rgb_arr.dtype,
+        "transform": Affine(image_resolution, 0, bbox[0], 0, -image_resolution, bbox[3]),
```

Comment:
I'm not sure why you have to define a transform, aren't the input geometry + the mask function enough to keep the geographical information?

Reply:
by the way, it looks like rasterio.mask has an out_transform output, maybe it would be better to use it instead

Reply:
Not sure, this is a cp-paste ^^'
I'll look into simplifications.

Reply:
I'm thinking this simplification can be made in another PR. Here I simply reuse the code from the BDOrthosVintageExtractor for the LAZExtractor. Refactoring the patch creation is not directly related. Additionnaly, I'm not very familiar with all the ways to write a raster so I'd rather let you make the modification you think is best in another PR. :)